src/fswAlgorithms/attControl/rateControl/rateControl.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ ISC License
+
+ Copyright (c) 2025, Laboratory for Atmospheric and Space Physics, University of Colorado at Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#include "rateControl.h"
+
+#include "architecture/utilities/avsEigenSupport.h"
+#include "architecture/utilities/linearAlgebra.h"
+
+/*! This method performs a complete reset of the module.  Local module variables that retain
+ time varying states between function calls are reset to their default values.
+ @return void
+ @param callTime [ns] Time the method is called
+*/
+void RateControl::reset(uint64_t callTime) {
+    // Check if the required input messages are linked
+    if (!this->guidInMsg.isLinked()) {
+        _bskLog(this->bskLogger, BSK_ERROR, "mrpProportionalDerivative.guidInMsg wasn't connected.");
+    }
+    if (!this->vehConfigInMsg.isLinked()) {
+        _bskLog(this->bskLogger, BSK_ERROR, "mrpProportionalDerivative.vehConfigInMsg wasn't connected.");
+    }
+
+    // Read the VehicleConfigMsgPayload input message
+    if (this->vehConfigInMsg.isWritten()) {
+        VehicleConfigMsgPayload vcInMsg;
```

Comment:
:small_red_triangle_down: You can declare and assign in a single line. `VehicleConfigMsgPayload vcInMsg = this->vehConfigInMsg();`